### ISSUE TITLE
iPython notebook nbconvert jpeg images

### ISSUE THREAD

**Issue by Aradan** [ipython/ipython#8340]
Hi,
when I try to convert a notebook that contains a `jpeg` or `jpg` image, I get the following error:

    ! LaTeX Error: Unknown graphics extension: .jpe.

Is the `nbconvert` system restricting the file extension?
In the `*_files` folder the extension looks right

**Comment by takluyver** [comment]
What does this give you?

```python
from mimetypes import guess_extension
guess_extension('image/jpeg')
```

**Comment by Aradan** [comment]
This

    In [64]: guess_extension('image/jpeg')
    Out[64]: '.jpe'

**Comment by takluyver** [comment]
I would guess that's the problem - I get `.jpg` for that. Something on your system reckons that JPEG images should have a .jpe extension, and Latex doesn't recognise that. But you say it's saving them with the right extension (I assume you mean .jpg) in the `*_files` directory?

**Comment by Aradan** [comment]
Yes, if I have a `jpg` or `jpeg` in the notebook, they are converted to `jpeg`
but Latex fails with the `jpe` error

**Comment by takluyver** [comment]
That's bizarre, it should use the same filename in both places. Is it possible that Latex still thinks file extensions can't be more than 3 characters?

**Comment by Aradan** [comment]
Yep, it is very strange. I don't think LaTeX rejects a `jpeg` file. What I do not understand is why the `nbconvert` would generate a `jpeg` file for both, a `jpg` or `jpeg`, and also why this format changes again when writing the `.tex` file. Is it `guess_extension` giving like a random choice of all the possibilities for the `jpg` format?

**Comment by takluyver** [comment]
I'm not sure what you mean by generating a jpeg file for both jpg and jpeg - where are the extensions coming from? It's possible that there's something non-deterministic in `guess_extension()`, but it should only be calling that once per output image, so what it puts in the latex file should always match the filename it saves.

**Comment by Aradan** [comment]
I mean that, when i use a `jpg` file in my notebook, I get a `jpeg` in the `*_files` directory and the `jpe` error in Latex.
The same occurs if I use a `jpeg` in my notebook, which is why I don't know how the `nbconvert` system is generating the files.

**Comment by takluyver** [comment]
How do you use those files? If the image data is getting embedded in the notebook, then nbconvert doesn't know about the original extension you fetched the image from, it just sees an image/jpeg mimetype.

Can you run nbconvert with `--to latex` and see what the relevant snippet of Latex code for an image looks like?

**Comment by Aradan** [comment]
Now in the `*_files` folder I always get the `jpe` file, which is the one written to the Latex file (I don't know if it is because I removed the old `*_files` folder)
The `tex` file looks like this:

   A general view of the system is depicted in the following figure:

    \begin{Verbatim}[commandchars=\\\{\}]{\color{incolor}In [{\color{incolor}96}]:} \PY{n}{Image}\PY{p{(}\PY{l+s}{\PYZsq{}}\PY{l+s}{figs/nanotube\PYZus{}scheme.jpeg}\PY{l+s{\PYZsq{}}\PY{p}{)}
    \end{Verbatim}
    \texttt{\color{outcolor}Out[{\color{outcolor}96}]:}
    
    \begin{center}
    \adjustimage{max size={0.9\linewidth}{0.9\paperheight}}{Data_Analysis_files/Data_Analysis_10_0.jpe}
    \end{center}
    { \hspace*{\fill} \\}
 
Latex complains because it doesn't know about `jpe`

Thanks for your help :)

**Comment by takluyver** [comment]
OK, that's making sense now. When we ask for the file extension for JPEG files, the system tells us it's `.jpe`, we use that, and Latex doesn't recognise it. What platform are you on? I know we've had some trouble with mime types on Windows before, though I think that was going the other way (extension -> mime).

**Comment by Aradan** [comment]
I'm using Ubuntu 14.04 and the latest IPython version

**Comment by cing** [comment]
I'm getting this error on OSX 10.11 with the latest full MacTex installation, with the latest jupyter from conda and the current development version of nbconvert (4.2.0-dev). First, here's the source of the notebook with data redacted:
```
{                                                                                                                            
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },  
   "outputs": [
    {   
     "data": { "image/jpeg": "/9j/4AAQSkZJRgABAQAAAQABAAD/2wBDA..." },
      "text/plain": [                                                                                                        
       "<IPython.core.display.Image object>"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from IPython.display import Image\n",
    "Image(filename='LysozymeStructure.jpg')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
```

As you can see, the image is a jpg. But when I run this, I get the error above about "Unknown graphics extension: .jpe".

```
jupyter nbconvert --to pdf TestingJPGtoPDF.ipynb
```

The following gives me .jpe,
```
> from mimetypes import guess_extension
> guess_extension('image/jpeg')
'.jpe'
```

Interestingly, if make it a Markdown cell and embed the same image, Save to PDF works fine.
```
{                                                                                                                            
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },  
   "source": [
    "![](LysozymeStructure.jpg)"
   ]   
  },  
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },  
   "outputs": [], 
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2", 
   "language": "python",
   "name": "python2"
  },  
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },  
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
```

**Comment by bkasel** [comment]
I have the same problem on Ubuntu 14.04.

**Comment by takluyver** [comment]
This should be fixed by jupyter/nbconvert#175. If you can still reproduce it with nbconvert master, please file a new bug on that repo.
